project-backend/Controllers/BidController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using project_backend.Models.BidController.AddBid;
+using project_backend.Models.JobController.AddJob;
+using project_backend.Providers.BidProvider;
+using System.Linq;
+using project_backend.Models.Utils;
+using Microsoft.AspNetCore.Http;
+using project_backend.Models.BidController.EditBid;
+using project_backend.Models.Exceptions;
+using project_backend.Models.Bid;
+using project_backend.Models;
+
+namespace project_backend.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class BidController : ControllerBase
+    {
+        private readonly ILogger<BidController> _logger;
+        private readonly IBidProvider _bidProvider;
+
+        public BidController(ILogger<BidController> logger, IBidProvider bidProvider)
+        {
+            _logger = logger;
+            _bidProvider = bidProvider;
+        }
+
+        [Authorize]
+        [HttpPost]
+        [ProducesResponseType(typeof(BidDAO), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(Error), StatusCodes.Status401Unauthorized)]
+        public IActionResult AddBid([FromBody] AddBidQueryObject bid)
+        {
+            var userIdClaim = HttpContext.User.GetUserIdClaim();
+            BidDAO newBid;
+            try
+            {
+                newBid = _bidProvider.AddBid(bid.Sum, int.Parse(userIdClaim.Value), bid.JobId);
+            }
+            catch (NotQualifiedException ex)
+            {
+                return Unauthorized(new Error(ex.Message));
```

Comment:
This should be Forbidden rather than Unauthorized. Unauthorized is used when the server does not know the identity of the user, forbidden means the server knows the user, but he is not allowed to access the resource he's trying to

Reply:
I can see what you mean. A bit counter intuitive because it sounds like you are describing something in the lines of unauthenticated. Usually authorization involves permission problems, not identity ones, so I thought that unauthorized might be ok. In terms of error codes, you are completely right.

Reply:
Should be good now. Can't pass the Forbid any object so it will be a plain IActionResult